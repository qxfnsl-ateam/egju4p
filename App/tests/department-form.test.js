import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import DepartmentForm from '../app-src/js/department-form';



describe('<DepartmentForm />', () => {
	it('Render new unit creation', () => {
		const cmp = shallow(
			<DepartmentForm onUpdate={jest.fn()} onDelete={jest.fn()} />
		);

		const eId = cmp.find('input[type="hidden"][name="id"]');
		expect(eId.length).toBe(0);

		const eName = cmp.find('input[type="text"][name="name"]');
		expect(eName.length).toBe(1);
		expect(eName.prop('defaultValue')).toBeFalsy();
	});


	it('Render unit editing', () => {
		const entry = {id:7, name:'Department 7'};
		const cmp = shallow(
			<DepartmentForm entry={entry} onUpdate={jest.fn()} onDelete={jest.fn()} />
		);

		const eId = cmp.find('input[type="hidden"][name="id"]');
		expect(eId.length).toBe(1);
		expect(eId.prop('value')).toBe(entry.id);

		const eName = cmp.find('input[type="text"][name="name"]');
		expect(eName.length).toBe(1);
		expect(eName.prop('defaultValue')).toMatch(entry.name);
	});


	it('Clickability', () => {
		const entry = {id:7, name:'Department 7'};
		const onUpdate = jest.fn();
		const onDelete = jest.fn();
		const cmp = mount(
			<DepartmentForm entry={entry} onUpdate={onUpdate} onDelete={onDelete} />
		);

		const eUpdate = cmp.find('button.staff-form-update');
		expect(eUpdate.length).toBe(1);
		expect(eUpdate.prop('onClick')).toBeDefined();

		const eDelete = cmp.find('button.staff-form-delete');
		expect(eDelete.length).toBe(1);
		expect(eDelete.prop('onClick')).toBe(onDelete);

		const evt = {target:{parentNode:{reportValidity:() => true}}};
		eUpdate.simulate('click', evt);
		expect(onUpdate).toBeCalled();
		eDelete.simulate('click', evt);
		expect(onDelete).toBeCalled();
	});


	it('DOM tree', () => {
		const entry = {id:7, name:'Department 7'};
		const cmp = renderer.create(
			<DepartmentForm entry={entry} onUpdate={jest.fn()} onDelete={jest.fn()} />
		);

		expect(cmp.toJSON()).toMatchSnapshot();
	});
});