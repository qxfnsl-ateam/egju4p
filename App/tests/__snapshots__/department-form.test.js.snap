// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DepartmentForm /> DOM tree 1`] = `
<form
  className="staff-form staff-form-dep"
>
  <h4
    className="staff-form-title text-primary"
  >
    Изменить информацию об отделе
  </h4>
  <input
    name="id"
    type="hidden"
    value={7}
  />
  <div
    className="form-group"
  >
    <label
      className="staff-dep-name"
      htmlFor="staff-dep-name"
    >
      Название
    </label>
    <input
      className="form-control"
      defaultValue="Department 7"
      id="staff-dep-name"
      name="name"
      pattern="[^\\\\\\\\s]{3,}"
      required={true}
      type="text"
    />
  </div>
  <button
    className="staff-form-update btn btn-default"
    onClick={[Function]}
  >
    Сохранить
  </button>
  <button
    className="staff-form-delete btn btn-danger"
    onClick={[Function]}
  >
    Удалить
  </button>
  <button
    className="staff-form-reset btn btn-default"
    type="reset"
  >
    Сбросить
  </button>
</form>
`;
